### PR TITLE
Week one's task

### PR DESCRIPTION
**Pull request by odelolajosh**

### CODE REVIEW

File: intermediate-week-1/controllers/auth/login.js [hunk 1]
```diff
@@ -1,10 +1,35 @@
...
+
+const login = async (req, res) => {
   /*send a response with the following format if the login is successful
     *{
         accessToken: **********,
         refreshToken: *********
     }
     */
+  const { password, email } = req.body;
+  try {
```
Comment (Ose-4g) [inline intermediate-week-1/controllers/auth/login.js:13]: Nice one. Handling errors with a try catch block.

File: intermediate-week-1/controllers/auth/login.js [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
+    const name = user.firstName + ' ' + user.lastName;
+    const accessToken = generateAccessToken(user._id, user.email, name);
+    const refreshToken = generateRefreshToken(user._id, user.email, name);
+
+    res.status(200).json({
+      accessToken,
+      refreshToken,
+    });
+  } catch (err) {
+    res.status(501).json({
```
Comment (Ose-4g) [inline intermediate-week-1/controllers/auth/login.js:29]: Nice error handling. 

There is a more graceful way to handle errors though. 

Check the NodeJs course in the resources. 

Well done.

File: intermediate-week-1/controllers/auth/refresh-token.js [hunk 3]
```diff
@@ -12,6 +14,32 @@ const refreshAccessToken = async (req, res, next) => {
...
+    const { payload: refresh, expired } = verifyRefreshToken(refreshToken || '');
+    if (!refresh) {
+      res.status(403).send('Invalid token!!');
+    }
+
+    if (expired) {
+      res.status(403).send('Expired token!!');
+    }
+
+    const { _id, email, name } = refresh;
```
Comment (Ose-4g) [inline intermediate-week-1/controllers/auth/refresh-token.js:30]: Nice work.

File: intermediate-week-1/controllers/auth/signup.js [hunk 4]
```diff
@@ -12,9 +14,17 @@ const signup = async (req, res, next) => {
...
+  try {
+    const { firstName, lastName, email, password, passwordConfirm } = req.body;
+    const user = new User({ firstName, lastName, email, password, passwordConfirm });
+    await user.save();
+
+    res.status(201).json({
+      message: 'Stored user details successfully',
+    });
+  } catch (err) {
+    res.status(501).send('Could not store user');
```
Comment (Ose-4g) [inline intermediate-week-1/controllers/auth/signup.js:26]: Nice error handling as well. 

In practice its good for all success responses and error responses to follow the same format. 

Thats why I recomment checking out the material on error handling.

File: intermediate-week-1/controllers/user/getUserInfo.js [hunk 5]
```diff
@@ -8,6 +8,21 @@ const getUserInfo = async (req, res, next) => {
    *      email: *****************
    * }
    */
+  try {
```
Comment (Ose-4g) [inline intermediate-week-1/controllers/user/getUserInfo.js:11]: Nice work here as well

File: intermediate-week-1/middlewares/auth.js [hunk 6]
```diff
@@ -1,9 +1,28 @@
...
+    /**
+     * verify the token and verify if user is logged in in or not
+     * If user is logged in then call the next() function to go to the next middleware
+     */
+    if (token && bearer === 'Bearer') {
+      const { payload } = verifyAccessToken(token);
+      if (!payload) throw new Error('Invalid Token');
+
+      req.user = await User.findById({ _id: payload._id });
+      return next();
```
Comment (Ose-4g) [inline intermediate-week-1/middlewares/auth.js:18]: Nice work here. 

But I'm not sure you should use "return". Just to be safe. 

Only use return when sending errors or when sending final succes response.

File: intermediate-week-1/utils/tokenUtils.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+const jwt = require('jsonwebtoken');
+
+const generateToken = (payload, privateKey, expiresIn) => jwt.sign(payload, privateKey, { expiresIn });
```
Comment (Ose-4g) [inline intermediate-week-1/utils/tokenUtils.js:3]: Reall good work here Chief. 

Well done.

**Comment by Ose-4g** [comment]
Really good work. 

The logic is clean and it shows you understand the code well. 

I also like the fact that your codebase is clean as well. i.e well formated and structured code. 

Well done.

File: intermediate-week-1/controllers/auth/login.js (hunk 2 above)
Comment (odelolajosh) [reply intermediate-week-1/controllers/auth/login.js:29]: I'll check it out

**Pull request closed without merging**